packages/query/src/bitcoin/bitcoin-client.ts

```diff
@@ -0,0 +1,292 @@
+import { HIRO_API_BASE_URL_MAINNET } from '@leather-wallet/constants';
+import axios from 'axios';
+
+import type { Paginated } from '../../types/api-types';
+import { useLeatherNetwork } from '../leather-query-provider';
+
+class Configuration {
+  constructor(public baseUrl: string) {}
+}
+
+export interface UtxoResponseItem {
+  txid: string;
+  vout: number;
+  status: {
+    confirmed: boolean;
+    block_height: number;
+    block_hash: string;
+    block_time: number;
+  };
+  value: number;
+}
+
+export interface UtxoWithDerivationPath extends UtxoResponseItem {
+  derivationPath: string;
+}
+
+interface BestinslotInscription {
+  inscription_name: string | null;
+  inscription_id: string;
+  inscription_number: number;
+  metadata: any | null;
+  wallet: string;
+  mime_type: string;
+  media_length: number;
+  genesis_ts: number;
+  genesis_height: number;
+  genesis_fee: number;
+  output_value: number;
+  satpoint: string;
+  collection_name: string | null;
+  collection_slug: string | null;
+  last_transfer_block_height: number;
+  content_url: string;
+  bis_url: string;
+  byte_size: number;
+}
+
+export interface BestinslotInscriptionByIdResponse {
+  data: BestinslotInscription;
+  block_height: number;
+}
+
+export interface BestinslotInscriptionsByTxIdResponse {
+  data: { inscription_id: string }[];
+  blockHeight: number;
+}
+
+interface Brc20TokenResponse {
+  ticker: string;
+  overall_balance: string;
+  available_balance: string;
+  transferrable_balance: string;
+  image_url: string | null;
+}
+
+export interface Brc20Token extends Brc20TokenResponse {
+  decimals: number;
+  holderAddress: string;
+}
+
+interface Brc20TokenTicker {
+  id: string;
+  number: number;
+  block_height: number;
+  tx_id: string;
+  address: string;
+  ticker: string;
+  max_supply: string;
+  mint_limit: string;
+  decimals: number;
+  deploy_timestamp: number;
+  minted_supply: string;
+  tx_count: number;
+}
+
+interface Brc20TickerResponse {
+  data: Brc20TokenTicker;
+  block_height: number;
+}
+
+interface BestinslotBrc20AddressBalanceResponse {
+  block_height: number;
+  data: Brc20TokenResponse[];
+}
+
+class BestinslotApi {
+  url = 'https://api.bestinslot.xyz/v3';
+  private defaultOptions = {
+    headers: {
+      'x-api-key': `${process.env.BESTINSLOT_API_KEY}`,
+    },
+  };
+  constructor(public configuration: Configuration) {}
```

Comment:
I think we should move away from the pattern of wrapping APIs in big classes. 

It isn't beneficial vs flat functions, and the only reason we do this is because that's how the auto-generated API from Hiro looks.